### ISSUE TITLE
[1.8.2-beta.18] "Unexpected token  in JSON" at build 

### ISSUE THREAD

**Issue by houshuang** [meteor/meteor#10688]
Mac OSX 10.14.6, Meteor 1.8.2-beta.18, when compiling, I get an unexpected token in JSON at position 0. It works fine with 1.8.1.

You can try this in https://github.com/chili-epfl/FROG, branch upgrade1.18-beta.18 (branch develop is on 1.8.1 otherwise identical). First run ./initial_setup.sh, then meteor from the frog folder.

```js
SyntaxError: Unexpected token  in JSON at position 0
    at JSON.parse (<anonymous>)
    at makeOptimistic (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/tools/fs/tools/fs/optimistic.ts:257:17)
    at wrap.makeCacheKey (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/tools/fs/tools/fs/optimistic.ts:35:15)
    at recomputeNewValue (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/optimism/src/entry.ts:177:31)
    at Slot.withValue (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/@wry/context/lib/context.js:73:29)
    at reallyRecompute (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/optimism/src/entry.ts:155:19)
    at Entry.recompute (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/optimism/src/entry.ts:92:9)
    at optimistic (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/optimism/src/index.ts:112:25)
    at wrap (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/tools/fs/tools/fs/optimistic.ts:298:19)
    at recomputeNewValue (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/optimism/src/entry.ts:177:31)
    at Slot.withValue (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/@wry/context/lib/context.js:73:29)
    at reallyRecompute (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/optimism/src/entry.ts:155:19)
    at Entry.recompute (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/optimism/src/entry.ts:92:9)
    at optimistic (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/optimism/src/index.ts:112:25)
    at find (/tools/isobuild/package-source.js:1286:25)
    at subdirectories.forEach.subdir (/tools/isobuild/package-source.js:1333:27)
    at Array.forEach (<anonymous>)
    at find (/tools/isobuild/package-source.js:1312:22)
    at find (/tools/isobuild/package-source.js:1344:25)
    at files.withCache (/tools/isobuild/package-source.js:1356:34)
    at Object.withCache (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/tools/fs/tools/fs/files.ts:1646:18)
    at PackageSource._findSources (/tools/isobuild/package-source.js:1356:18)
    at SourceArch.getFiles (/tools/isobuild/package-source.js:912:32)
    at /tools/isobuild/compiler.js:406:23
    at files.withCache (/tools/isobuild/compiler.js:186:28)
    at Object.withCache (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/tools/fs/tools/fs/files.ts:1646:18)
    at /tools/isobuild/compiler.js:185:11
    at Function._.each._.forEach (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:186:9)
    at Object.compile (/tools/isobuild/compiler.js:180:5)
    at /tools/isobuild/bundler.js:3231:24
    at Object.capture (/tools/utils/buildmessage.js:283:5)
    at bundle (/tools/isobuild/bundler.js:3178:31)
    at files.withCache (/tools/isobuild/bundler.js:3123:32)
    at Slot.withValue (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/@wry/context/lib/context.js:73:29)
    at Object.withCache (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/tools/fs/tools/fs/files.ts:1646:39)
    at Object.bundle (/tools/isobuild/bundler.js:3123:16)
    at Profile.run (/tools/runners/run-app.js:572:24)
    at Function.run (/Users/stian/.meteor/packages/meteor-tool/.1.8.2-beta.18.1baxltq.rgxxk++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/tools/tool-env/tools/tool-env/profile.ts:289:14)
    at bundleApp (/tools/runners/run-app.js:571:34)
    at AppRunner._runOnce (/tools/runners/run-app.js:613:35)
    at AppRunner._fiber (/tools/runners/run-app.js:929:28)
    at /tools/runners/run-app.js:401:12
```

**Comment by KoenLav** [comment]
Same here.

**Comment by benjamn** [comment]
This should be fixed if you run `meteor update --release 1.8.2-rc.0`, now that we're using `json5` to read JSON files. If the problem persists, there's a chance we're still using the default `JSON.parse` API to read some JSON files, so please see if you can determine which file is causing the problem, or at least where in the `meteor/tools` codebase the `JSON.parse` is happening.

**Comment by benjamn** [comment]
Unfortunately, I had to revert the `json5` changes because they introduced a fairly severe performance regression: 0e19c365a8ee115db5b45e69a9f9173dd89cb29a

However, I believe I found another way to fix this issue, which will be released in `1.8.2-rc.1`: df3fac2e1f972b3920b40e3f50db65770a1b53ce
